lib/cdo/aws/cloud_formation.rb

```diff
@@ -138,32 +143,53 @@ def update_bootstrap_script
         )
       end
 
+      # Prints the latest output from a CloudWatch Logs log stream, if present.
+      def tail_log(quiet: false)
+        log_config = {
+          log_group_name: STACK_NAME,
+          log_stream_name: LOG_NAME
+        }
+        log_config[:next_token] = @@log_token unless @@log_token.nil?
+        # Return silently if log doesn't exist.
+        resp = logs.get_log_events(log_config) rescue return
+        resp.events.each do |event|
+          CDO.log.info(event.message) unless quiet
+        end
+        @@log_token = resp.next_forward_token
+      end
+
+      # Prints the latest CloudFormation stack events.
+      def tail_events(stack_id)
+        stack_events = cfn.describe_stack_events(stack_name: stack_id).stack_events
+        stack_events.reject{|event| event.timestamp <= @@event_timestamp}.sort_by(&:timestamp).each do |event|
+          CDO.log.info "#{event.timestamp}- #{event.logical_resource_id} [#{event.resource_status}]: #{event.resource_status_reason}"
+        end
+        @@event_timestamp = stack_events.map(&:timestamp).max
+      end
+
       def wait_for_stack(action, start_time)
         CDO.log.info "Stack #{action} requested, waiting for provisioning to complete..."
+        stack_id = STACK_NAME
         begin
-          cfn.wait_until("stack_#{action}_complete".to_sym, stack_name: STACK_NAME) do |w|
+          @@event_timestamp = start_time
+          @@log_token = nil
+          tail_log(quiet: true)
+          stack_id = cfn.describe_stacks(stack_name: stack_id).stacks.first.stack_id
+          cfn.wait_until("stack_#{action}_complete".to_sym, stack_name: stack_id) do |w|
             w.max_attempts = 360 # 1 hour
```

Comment:
What units is this in? Is this meant to be `3600`? If `360` is intended, it is worth noting this.